pkg/tcpip/adapters/gonet/gonet.go

```diff
@@ -556,6 +556,17 @@ type PacketConn struct {
 	wq    *waiter.Queue
 }
 
+// NewPacketConn creates a new PacketConn.
+func NewPacketConn(s *stack.Stack, wq *waiter.Queue, ep tcpip.Endpoint) *PacketConn {
```

Comment:
Could you update DialUDP to use this constructor so that we don't duplicate the same code.

Reply:
Actually I am just doing it, don't worry about it.

Reply:
Thanks!